src/topology/homotopy/induced_maps.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+import topology.homotopy.product
+import topology.homotopy.equiv
+import topology.homotopy.casts
+import category_theory.equivalence
+
+/-!
+# Homotopic maps induce naturally isomorphic functors
+
+## Main definitions
+
+  - `fundamental_groupoid_functor.homotopic_maps_nat_iso H` The natural isomorphism
+    between the induced functors `f : π(X) ⥤ π(Y)` and `g : π(X) ⥤ π(Y)`, given a homotopy
+    `H : f ∼ g`
+
+  - `fundamental_groupoid_functor.equiv_of_homotopy_equiv hequiv` The equivalence of the categories
+    `π(X)` and `π(Y)` given a homotopy equivalence `hequiv : X ≃ₕ Y` between them.
+
+## Implementation notes
+  - In order to be more universe polymorphic, we define `continuous_map.homotopy.ulift_map`
+  which lifts a homotopy from `I × X → Y` to `(Top.of ((ulift I) × X)) → Y`
```

Comment:
Where do you need this universe polymorphism? It's a bit weird to introduce it halfway through the API and only introduce one variation of it: why not ulifting all of `I`, `X` and `Y`? And isn't it universe polymorphic already?